### PR TITLE
xcompile: use a host toolchain on M1 Macs too

### PR DESCRIPTION
**Pull request by benesch**
@guswynn found a newer set of cross-compiling toolchains for macOS that
support the aarch64 architecture. This makes using mzcompose on M1 Macs
much faster since compilations no longer occur in Docker for Mac, which
has painfully slow I/O.

### Motivation

  * This PR adds a known-desirable feature.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

File: misc/python/materialize/xcompile.py [hunk 1]
```diff
@@ -184,7 +180,7 @@ def _bootstrap_darwin_x86_64() -> None:
         contents = ""
     if contents == BOOTSTRAP_VERSION:
         return
-    spawn.runv(["brew", "install", "SergioBenitez/osxct/x86_64-unknown-linux-gnu"])
-    spawn.runv(["rustup", "target", "add", "x86_64-unknown-linux-gnu"])
+    spawn.runv(["brew", "install", f"messense/macos-cross-toolchains/{target(arch)}"])
```
Comment (guswynn) [inline misc/python/materialize/xcompile.py:183]: works for all cases! nice!

**Review by guswynn — APPROVED** [review]
YES

now all thats left is actually make kafka like the network in mzcompose!

File: misc/python/materialize/xcompile.py (hunk 1 above)
Comment (guswynn) [reply misc/python/materialize/xcompile.py:183]: oh wait, we dont need the rustup anymore?

Comment (benesch) [reply misc/python/materialize/xcompile.py:183]: I know right!

**Comment by benesch** [comment]
> now all thats left is actually make kafka like the network in mzcompose!

Er, well, it was segfaulting because of who knows what, right? :/ Probably not worth the time to look into.

File: misc/python/materialize/xcompile.py (hunk 1 above)
Comment (benesch) [reply misc/python/materialize/xcompile.py:183]: We do! It's still there?

Comment (guswynn) [reply misc/python/materialize/xcompile.py:180]: Oh I am bad at reading stuff on github apparently!

Comment (benesch) [reply misc/python/materialize/xcompile.py:180]: No worries, glad I'm not crazy!

**Pull request merged**